src/interface/viewport.c

```diff
@@ -2636,8 +2636,216 @@ void store_interaction_info(paint_struct *ps)
  *
  *  rct2: 0x00679074
  */
-void sub_679074(rct_drawpixelinfo *dpi, int imageId, int x, int y)
+void sub_679074(rct_drawpixelinfo *dpi, int imageId, sint16 x, sint16 y)
 {
+	rct_g1_element image = g1Elements[imageId & 0x7FFFF];
+
+	switch (dpi->zoom_level) {
+		case 0:
+			// TODO: loc_6790A0:
+			break;
+
+		case 1:
+			if (image.flags & 0x20) {
+				return;
+			}
+
+			if (!(image.flags & 0x10)) {
+				dpi->zoom_level--;
+				dpi->x /= 2;
```

Comment:
Use `>>= 1` otherwise there will be a tiny error with single pixels. `-1/2 = 0`. `-1 >> 1 = -1`.